diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/utils/FunctonASTNodeUtils.kt

```diff
@@ -1,11 +1,36 @@
 package org.cqfn.diktat.ruleset.utils
 
 import com.pinterest.ktlint.core.ast.ElementType
+import com.pinterest.ktlint.core.ast.ElementType.ANNOTATION_ENTRY
+import com.pinterest.ktlint.core.ast.ElementType.IDENTIFIER
+import com.pinterest.ktlint.core.ast.ElementType.MODIFIER_LIST
+import org.cqfn.diktat.ruleset.rules.PackageNaming.Companion.PACKAGE_PATH_ANCHOR
 import org.jetbrains.kotlin.com.intellij.lang.ASTNode
 import org.jetbrains.kotlin.kdoc.parser.KDocKnownTag
 import org.jetbrains.kotlin.psi.KtParameter
 import org.jetbrains.kotlin.psi.KtParameterList
 
+/**
+ * Checks whether function from this [ElementType.FUN] node is from test class based on annotations and file location
+ */
+fun ASTNode.isTestFun(filePathParts: List<String>): Boolean {
+    checkNode(this)
+    val hasTestAnnotation = findChildByType(MODIFIER_LIST)?.getAllChildrenWithType(ANNOTATION_ENTRY)
```

Comment:
I guess that there is no need to get MODIFIER_LIST, you can search directly for ANNOTATION_ENTRY. This type of node exists only in MODIFIER_LIST

Reply:
There could also be annotated parameters, for example. Also, annotation entries are not direct children of `FUN`, so we still need to specify depth in tree somehow.